'IS_CONSTANT_ARRAY' undeclared when installing 'runtime'

I getting this error when trying compile Zephir Runtime

    find . -name \*.gcno -o -name \*.gcda | xargs rm -f
    find . -name \*.lo -o -name \*.o | xargs rm -f
    find . -name \*.la -o -name \*.a | xargs rm -f
    find . -name \*.so | xargs rm -f
    find . -name .libs -a -type d|xargs rm -rf
    rm -f libphp.la       modules/* libs/*
    Cleaning..
    Configuring for:
    PHP Api Version:         20131106
    Zend Module Api No:      20131226
    Zend Extension Api No:   220131226
    checking for grep that handles long lines and -e... /bin/grep
    checking for egrep... /bin/grep -E
    checking for a sed that does not truncate output... /bin/sed
    checking for cc... cc
    checking whether the C compiler works... yes
    checking for C compiler default output file name... a.out
    checking for suffix of executables...
    checking whether we are cross compiling... no
    checking for suffix of object files... o
    checking whether we are using the GNU C compiler... yes
    checking whether cc accepts -g... yes
    checking for cc option to accept ISO C89... none needed
    checking how to run the C preprocessor... cc -E
    checking for icc... no
    checking for suncc... no
    checking whether cc understands -c and -o together... yes
    checking for system library directory... lib
    checking if compiler supports -R... no
    checking if compiler supports -Wl,-rpath,... yes
    checking build system type... x86_64-unknown-linux-gnu
    checking host system type... x86_64-unknown-linux-gnu
    checking target system type... x86_64-unknown-linux-gnu
    checking for PHP prefix... /usr
    checking for PHP includes... -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib
    checking for PHP extension directory... /usr/lib/php5/20131226
    checking for PHP installed headers prefix... /usr/include/php5
    checking if debug is enabled... no
    checking if zts is enabled... no
    checking for re2c... re2c
    checking for re2c version... 0.13.5 (ok)
    checking for gawk... gawk
    checking whether to enable zephir... yes, shared
    ./configure: line 4031: llvm-config-3.3: command not found
    ./configure: line 4032: llvm-config-3.3: command not found
    checking for stdc++... checking for g++... g++
    checking whether we are using the GNU C++ compiler... yes
    checking whether g++ accepts -g... yes
    yes
    checking whether we are using the GNU C++ compiler... (cached) yes
    checking whether g++ accepts -g... (cached) yes
    checking how to run the C++ preprocessor... g++ -E
    checking how to print strings... printf
    checking for a sed that does not truncate output... (cached) /bin/sed
    checking for fgrep... /bin/grep -F
    checking for ld used by cc... /usr/bin/ld
    checking if the linker (/usr/bin/ld) is GNU ld... yes
    checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
    checking the name lister (/usr/bin/nm -B) interface... BSD nm
    checking whether ln -s works... yes
    checking the maximum length of command line arguments... 1572864
    checking whether the shell understands some XSI constructs... yes
    checking whether the shell understands "+="... yes
    checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
    checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
    checking for /usr/bin/ld option to reload object files... -r
    checking for objdump... objdump
    checking how to recognize dependent libraries... pass_all
    checking for dlltool... no
    checking how to associate runtime and link libraries... printf %s\n
    checking for ar... ar
    checking for archiver @FILE support... @
    checking for strip... strip
    checking for ranlib... ranlib
    checking for gawk... (cached) gawk
    checking command to parse /usr/bin/nm -B output from cc object... ok
    checking for sysroot... no
    checking for mt... mt
    checking if mt is a manifest tool... no
    checking for ANSI C header files... yes
    checking for sys/types.h... yes
    checking for sys/stat.h... yes
    checking for stdlib.h... yes
    checking for string.h... yes
    checking for memory.h... yes
    checking for strings.h... yes
    checking for inttypes.h... yes
    checking for stdint.h... yes
    checking for unistd.h... yes
    checking for dlfcn.h... yes
    checking for objdir... .libs
    checking if cc supports -fno-rtti -fno-exceptions... no
    checking for cc option to produce PIC... -fPIC -DPIC
    checking if cc PIC flag -fPIC -DPIC works... yes
    checking if cc static flag -static works... yes
    checking if cc supports -c -o file.o... yes
    checking if cc supports -c -o file.o... (cached) yes
    checking whether the cc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
    checking whether -lc should be explicitly linked in... no
    checking dynamic linker characteristics... GNU/Linux ld.so
    checking how to hardcode library paths into programs... immediate
    checking whether stripping libraries is possible... yes
    checking if libtool supports shared libraries... yes
    checking whether to build shared libraries... yes
    checking whether to build static libraries... no
    checking how to run the C++ preprocessor... g++ -E
    checking for ld used by g++... /usr/bin/ld -m elf_x86_64
    checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
    checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
    checking for g++ option to produce PIC... -fPIC -DPIC
    checking if g++ PIC flag -fPIC -DPIC works... yes
    checking if g++ static flag -static works... no
    checking if g++ supports -c -o file.o... yes
    checking if g++ supports -c -o file.o... (cached) yes
    checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
    checking dynamic linker characteristics... (cached) GNU/Linux ld.so
    checking how to hardcode library paths into programs... immediate
    configure: creating ./config.status
    config.status: creating config.h
    config.status: executing libtool commands
    /bin/bash /home/vagrant/z/zephir/runtime/libtool --mode=compile cc  -I. -I/home/vagrant/z/zephir/runtime -DPHP_ATOM_INC -I/home/vagrant/z/zephir/runtime/include -I/home/vagrant/z/zephir/runtime/main -I/home/vagrant/z/zephir/runtime -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib  -DHAVE_CONFIG_H  -g -O2 -Wl,-rpath -O0 -g3 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS   -c
    /home/vagrant/z/zephir/runtime/zephir.c -o zephir.lo
    libtool: compile:  cc -I. -I/home/vagrant/z/zephir/runtime -DPHP_ATOM_INC -I/home/vagrant/z/zephir/runtime/include -I/home/vagrant/z/zephir/runtime/main -I/home/vagrant/z/zephir/runtime -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib -DHAVE_CONFIG_H -g -O2 -Wl,-rpath -O0 -g3 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -c /home/vagrant/z/zephir/runtime/zephir.c  -fPIC -DPIC -o .libs/zephir.o
    /bin/bash /home/vagrant/z/zephir/runtime/libtool --mode=compile cc  -I. -I/home/vagrant/z/zephir/runtime -DPHP_ATOM_INC -I/home/vagrant/z/zephir/runtime/include -I/home/vagrant/z/zephir/runtime/main -I/home/vagrant/z/zephir/runtime -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib  -DHAVE_CONFIG_H  -g -O2 -Wl,-rpath -O0 -g3 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS   -c
    /home/vagrant/z/zephir/runtime/kernel/main.c -o kernel/main.lo
    libtool: compile:  cc -I. -I/home/vagrant/z/zephir/runtime -DPHP_ATOM_INC -I/home/vagrant/z/zephir/runtime/include -I/home/vagrant/z/zephir/runtime/main -I/home/vagrant/z/zephir/runtime -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib -DHAVE_CONFIG_H -g -O2 -Wl,-rpath -O0 -g3 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -c /home/vagrant/z/zephir/runtime/kernel/main.c  -fPIC -DPIC -o kernel/.libs/main.o
    /bin/bash /home/vagrant/z/zephir/runtime/libtool --mode=compile cc  -I. -I/home/vagrant/z/zephir/runtime -DPHP_ATOM_INC -I/home/vagrant/z/zephir/runtime/include -I/home/vagrant/z/zephir/runtime/main -I/home/vagrant/z/zephir/runtime -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib  -DHAVE_CONFIG_H  -g -O2 -Wl,-rpath -O0 -g3 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS   -c
    /home/vagrant/z/zephir/runtime/kernel/memory.c -o kernel/memory.lo
    libtool: compile:  cc -I. -I/home/vagrant/z/zephir/runtime -DPHP_ATOM_INC -I/home/vagrant/z/zephir/runtime/include -I/home/vagrant/z/zephir/runtime/main -I/home/vagrant/z/zephir/runtime -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib -DHAVE_CONFIG_H -g -O2 -Wl,-rpath -O0 -g3 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -c /home/vagrant/z/zephir/runtime/kernel/memory.c  -fPIC -DPIC -o kernel/.libs/memory.o
    /home/vagrant/z/zephir/runtime/kernel/memory.c: In function 'zephirt_dtor_func':
    /home/vagrant/z/zephir/runtime/kernel/memory.c:576:14: error: 'IS_CONSTANT_ARRAY' undeclared (first use in this function)
             case IS_CONSTANT_ARRAY: {
                  ^
    /home/vagrant/z/zephir/runtime/kernel/memory.c:576:14: note: each undeclared identifier is reported only once for each function it appears in
    make: *** [kernel/memory.lo] Error 1


<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/20855219-is_constant_array-undeclared-when-installing-runtime?utm_campaign=plugin&utm_content=tracker%2F280146&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F280146&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/runtime/kernel/memory.c
+++ b/runtime/kernel/memory.c
@@ -572,8 +572,10 @@ static inline void zephirt_dtor_func(zval *zvalue ZEND_FILE_LINE_DC)
             CHECK_ZVAL_STRING_REL(zvalue);
             STR_FREE_REL(zvalue->value.str.val);
             break;
-        case IS_ARRAY:
-        case IS_CONSTANT_ARRAY: {
+#if PHP_VERSION_ID < 50600
+		case IS_CONSTANT_ARRAY:
+#endif
+        case IS_ARRAY: {
                 TSRMLS_FETCH();
                 if (zvalue->value.ht && (zvalue->value.ht != &EG(symbol_table))) {
                     /* break possible cycles */

